app/models/foreman_openscap/arf_report.rb

```diff
@@ -1,112 +1,118 @@
-require 'digest'
-
 module ForemanOpenscap
-  class ArfReport < ActiveRecord::Base
+  class ArfReport < ::Report
     include Taxonomix
+    extend Assetmix
+
+    RESULT = %w(pass fail error unknown notapplicable notchecked notselected informational fixed)
+    METRIC = %w(passed othered failed)
+    BIT_NUM = 10
+    MAX = (1 << BIT_NUM) - 1
 
-    belongs_to :asset
-    belongs_to :policy
+    belongs_to_asset
+    has_one :policy_arf_report, :dependent => :destroy
+    has_one :policy, :through => :policy_arf_report
     delegate :assetable, :to => :asset, :as => :assetable
-    has_many :xccdf_rule_results, :dependent => :destroy
-    has_one :arf_report_raw, :dependent => :destroy
-    has_one :arf_report_breakdown
     has_one :host, :through => :asset, :as => :assetable, :source => :assetable, :source_type => 'Host::Base'
 
     after_save :assign_locations_organizations
+    validate :result, :inclusion => { :in => RESULT }
 
     default_scope {
       with_taxonomy_scope do
-        order("foreman_openscap_arf_reports.created_at DESC")
+        order("reports.created_at DESC")
```

Comment:
please use `#{self.table_name}.created_at DESC` as we tend to rename tables from time to time